RecoMuon/GlobalTrackingTools/interface/MuonTrackingRegionByPtBuilder.h

```diff
@@ -0,0 +1,86 @@
+#ifndef RecoMuon_TrackingTools_MuonTrackingRegionByPtBuilder_H
+#define RecoMuon_TrackingTools_MuonTrackingRegionByPtBuilder_H
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "RecoTracker/TkTrackingRegions/interface/TrackingRegionProducer.h"
+#include "RecoTracker/TkTrackingRegions/interface/RectangularEtaPhiTrackingRegion.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+class MuonServiceProxy;
+class MeasurementTrackerEvent;
+class MagneticField;
+class IdealMagneticFieldRecord;
+class MultipleScatteringParametrisationMaker;
+class TrackerMultipleScatteringRecord;
+
+class MuonTrackingRegionByPtBuilder : public TrackingRegionProducer {
+public:
+  /// Constructor
+  explicit MuonTrackingRegionByPtBuilder(const edm::ParameterSet& par, edm::ConsumesCollector& iC) { build(par, iC); }
+  explicit MuonTrackingRegionByPtBuilder(const edm::ParameterSet& par, edm::ConsumesCollector&& iC) { build(par, iC); }
+
+  /// Destructor
+  ~MuonTrackingRegionByPtBuilder() override {}
+
+  /// Create Region of Interest
+  std::vector<std::unique_ptr<TrackingRegion> > regions(const edm::Event&, const edm::EventSetup&) const override;
+
+  /// Define tracking region
+  std::unique_ptr<RectangularEtaPhiTrackingRegion> region(const reco::TrackRef&) const;
+  std::unique_ptr<RectangularEtaPhiTrackingRegion> region(const reco::Track& t) const {
+    return region(t, *theEvent, *theEventSetup);
+  }
+  std::unique_ptr<RectangularEtaPhiTrackingRegion> region(const reco::Track&,
+                                                          const edm::Event&,
+                                                          const edm::EventSetup&) const;
+
+  /// Pass the Event to the algo at each event
+  void setEvent(const edm::Event&, const edm::EventSetup&);
+
+  /// Add Fill Descriptions
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  static void fillDescriptionsHLT(edm::ParameterSetDescription& descriptions);

```

Comment:
if you are not planning to do something like this https://github.com/cms-sw/cmssw/blob/f2bf00c7a78ce78a77ff7043d8f8a6b4eba8bf59/RecoMuon/GlobalTrackingTools/src/MuonTrackingRegionBuilder.cc#L268-L339 , you can remove it

Reply:
done.